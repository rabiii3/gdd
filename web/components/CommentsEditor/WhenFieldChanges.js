import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'react-final-form';
import { OnChange } from 'react-final-form-listeners';

const WhenFieldChanges = ({ field, becomes, set, to }) => (
  <Field name={set} subscription={{}}>
    {({ input: { onChange } }) => (
      <OnChange name={field}>
        {value => {
          if (value === becomes) {
            onChange(to)
          }
        }}
      </OnChange>
    )}
  </Field>
);

WhenFieldChanges.propTypes = {
  field: PropTypes.string,
  becomes: PropTypes.string,
  set: PropTypes.string,
  to: PropTypes.object,
}

export default WhenFieldChanges;