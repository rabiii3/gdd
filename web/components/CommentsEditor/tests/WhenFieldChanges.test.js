import React from 'react';
import { shallow } from 'enzyme';
import WhenFieldChanges from '../WhenFieldChanges';

describe('web | components | CommentsEditor | WhenFieldChanges', () => {
  describe('render', () => {
    it('should match snapshot', () => {
      
      const props = {
        field: 'test',
        becomes: 'none',
        set: 'result',
        to: {},
      };

      const wrapper = shallow(<WhenFieldChanges {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});