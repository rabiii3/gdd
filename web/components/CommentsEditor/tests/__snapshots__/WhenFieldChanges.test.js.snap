// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web | components | CommentsEditor | WhenFieldChanges render should match snapshot 1`] = `
<ReactFinalForm(Field)
  name="result"
  subscription={Object {}}
>
  <Component />
</ReactFinalForm(Field)>
`;
